llvm/include/llvm/Demangle/ItaniumDemangle.h

```diff
@@ -6176,6 +6180,10 @@ struct ManglingParser : AbstractManglingParser<ManglingParser<Alloc>, Alloc> {
                                Alloc>::AbstractManglingParser;
 };
 
+inline void OutputBuffer::printLeft(const Node &N) { N.printLeft(*this); }
```

Comment:
@zygoloid is this what you had in mind? I moved the `TrackingOutputBuffer` into LLDB now. That's where all the tracking logic would live. It's a bit unfortunate we have to duplicate some of the printing logic there (though at least we're only doing it for a couple of nodes)

Reply:
Thanks, yes. I agree that the duplication is not ideal, and I think it'd be fine to add a few more virtual function calls if it reduces the duplication, but this extension to the demangler looks nice and clean now.